R-packages/covidcast/R/wrangle.R

```diff
@@ -0,0 +1,296 @@
+    warn(paste("More than one `geo_type` present. Are you sure you want to",
+               "aggregate?"), class = "aggregate_signals_geo_type")
+  }
+
+  # Wide format
+  if (format == "wide") {
+    x = lapply(x, latest_issue) # Grab only the latest issue in wide form!
+    
+    # Rename value columns according to data source and signal combos, and drop
+    # a bunch of columns 
+    for (i in 1:N) {
+      src = x[[i]]$data_source[1]
+      sig = x[[i]]$signal[1]
+      x[[i]] = x[[i]] %>%
+        dplyr::rename_with(~ paste0(.x, ":", src, "_", sig),
+                           dplyr::starts_with("value")) %>%
+        dplyr::select(-c(data_source, signal, issue, lag, stderr, sample_size)) 
+    }
+    
+    # This is the wide data frame that we will eventually return
+    y = x[[1]]
+
+    # Pairwise full joins (if there's any pairs to join at all)
+    if (length(x) > 1) {
+      for (i in 2:length(x)) {
+        y = dplyr::full_join(y, x[[i]], by = c("geo_value", "time_value"))
+      }
+    }
+
+    # Set covidcast_signal_wide class, attributes, and return
+    class(y) = c("covidcast_signal_wide", "data.frame")
+    attributes(y)$metadata = meta_all
+    return(y)
+  }
+
+  # Long format
+  if (format == "long") {
+    # Pivot time-shifted value columns to long format, add column dt 
+    for (i in 1:N) {
+      x[[i]] = x[[i]] %>%
+        tidyr::pivot_longer(dplyr::starts_with("value"),
+                            names_to = "dt", values_to = "value") %>%
+        dplyr::mutate(dt = as.numeric(sub("value", "", dt)))
+    }
+
+    # Bind all rows together from the various data frames
+    y = do.call(dplyr::bind_rows, x)
+
+    # Set covidcast_signal_wide class, attributes, and return
+    class(y) = c("covidcast_signal_long", "data.frame")
+    attributes(y)$metadata = meta_all
+    return(y)
+  }
+}
+
+# Pivot covidcast_signal_wide object to "long" format
+#
+#' @export
+covidcast_longer = function(x) {
+  if (!("covidcast_signal_wide" %in% class(x))) {
+    stop("`x` must be a `covidcast_signal_wide` object.")
+  }
+  
+  # First pivot into long format
+  x = x %>%
+    tidyr::pivot_longer(dplyr::starts_with("value"),
+                        names_to = "dt_data_source_signal",
+                        values_to = "value") %>%
+    tidyr::separate(col = "dt_data_source_signal",
+                    into = c("dt", "data_source_signal"),
+                    sep = ":") %>%
+    tidyr::separate(col = "data_source_signal",
+                    into = c("data_source", "signal"),
+                    sep = "_", extra = "merge") 
+
+  # Now add dt column, and reorder columns a bit
+  x = x %>% dplyr::mutate(dt = as.numeric(sub("value", "", dt))) %>%
+    dplyr::relocate(data_source, signal, geo_value, time_value) %>%
+    dplyr::relocate(dt, .before = value)
+
+  # Change class and return
+  class(x) = c("covicast_signal_long", "data.frame")
+  return(x)
+}
+
+#' Pivot covidcast_signal_long object to "wide" format
+#
+#' @export
```

Comment:
Is this sufficient documentation for this function (and its counterpart `covidcast_longer`)?

Reply:
Thanks, I added a bit more.  This doesn't need extensive documentation because it just acts as if `aggregate_signals()` had been called with a different option, which I now explained.